src/Compiler/FSComp.txt

```diff
@@ -1775,3 +1775,4 @@ featureParsedHashDirectiveArgumentNonString,"# directives with non-quoted string
 3869,featureParsedHashDirectiveUnexpectedInteger,"Unexpected integer literal '%d'."
 3869,featureParsedHashDirectiveUnexpectedIdentifier,"Unexpected identifier '%s'."
 featureEmptyBodiedComputationExpressions,"Support for computation expressions with empty bodies: builder {{ }}"
+3870,parsExpectingUnionCaseField,"Expecting union case field"
```

Comment:
Apologies for being annoying here (👀) - just want to suggest a more strict wording here, because a case like
```fsharp
type U =
    | A of * int * int
```
we could fix by either adding a field name or adjusting a type:
```fsharp
type U1 =
    | A of a: int * int

type U2 =
    | A of int * int
```

I'd suggest a message like:
```suggestion
3870,parsExpectingUnionCaseField,"Expecting union case field name or type"
```

Reply:
@psfinaki Thank you for the suggestion, but I disagree.

> we could fix by either adding a field name or adjusting a type:

Both possible fixes are not about adding the expected field. The first one _removes_ the separator and modifies another field. The second one just removes the possible second field by removing the separator.

I do agree that the wording is not the most human-like, but I try to do it in-line with other parser errors, and this wording is both correct and similar to the generated and other hand-written ones. It's produced when a field separator is parser before an actual field, so it's very common to say that a field was expected.

Reply:
Alright, that's convincing, let's keep it as it is.